### DISCUSSION TITLE
Spawning Async methods

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#11096] [General - Components / Directives / etc] [answered]
How can I spawn two async methods from within an action. I've trimmed down the code to the important bits. I'll explain below. 

Currently my vue file has a method **clickedItem** which triggers 3 dispatches. The first dispatch sets the selected item in the 'state'. The other two dispatches follow by collecting data for the newly selected item.

What I was hoping to do was place the two '**fetchData**' and '**fetchDetails**' inside of the '**setSelection**' action so they would automatically trigger anytime the selection is set within the app. However when I do that, I noticed they no longer run in parallel, they appear to be blocking, running one after the other instead. How would I fix that?

**MyComponent.vue**

```
<template>
...
</template>

<script>
...

  setup(props, { emit }) {
    ...
    const clickedItem = (event) => {
      store.dispatch("app/setSelection", {isSelected: !props.item.isSelected,});
      store.dispatch("app/fetchData")
      store.dispatch("app/fetchDetails")
    };
    return {
      ...
    };
  },
});
</script>

```

I tried this but the two 'fetch' methods no longer run in parallel, which is what i want. They now seem to block each other.

**actions.js**

```
export async function setSelection(
  { commit, state, dispatch },
  payload
) {
  // code to set the tile selection (payload)
  ...

  // dispatch...
  await dispatch("fetchData");
  await dispatch("fetchDetails");
}
```

**Comment by hawkeye64** [comment, marked as answer]
Your first code block is asynchronous, your second is not.

> I noticed they no longer run in parallel

There are no guarantees when running asynchronously. They may appear blocking, but that's the nature of the beast. It really depends on the code within each one.

**Reply by JokerMartini** [reply]
fixed...
```
  dispatch("fetchData");
  dispatch("fetchDetails");
```
